pycbc/inference/models/hierarchical.py

```diff
@@ -723,24 +743,47 @@ def total_loglr(self):
             # not using self.primary_model.current_params, because others_model
             # may have its own static parameters
             current_params_other = other_model.current_params.copy()
-            for i in range(nums):
+            if not self.static_margin_params_in_other_models:
+                for i in range(nums):
+                    current_params_other.update(
+                        {key: value[i] if isinstance(value, numpy.ndarray)
+                         else value for key, value in margin_params.items()})
+                    other_model.update(**current_params_other)
+                    other_model.return_sh_hh = True
+                    sh_other, hh_other = other_model.loglr
+                    sh_others[i] += sh_other
+                    hh_others[i] += hh_other
+                    other_model.return_sh_hh = False
+                    # set logr, otherwise it will store (sh, hh)
+                    setattr(other_model._current_stats, 'loglr',
```

Comment:
My understanding is that when `pycbc_inference_model_stats` check `for pi, p in enumerate(model.default_stats):` it will try to access submodel's `loglr`, no? If so, I need to store it. When I not rest it, I found `other_model._current_stats` also contain (sh, hh) for each point.